Simplenote/src/main/java/com/automattic/simplenote/utils/IconResizer.java

```diff
@@ -0,0 +1,101 @@
+package com.automattic.simplenote.utils;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PaintFlagsDrawFilter;
+import android.graphics.PixelFormat;
+import android.graphics.Rect;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.PaintDrawable;
+
+/**
+ * Utility class to resize icons to match default icon size.
+ */
+public class IconResizer {
+
+    // Code is borrowed from com.android.launcher.Utilities.
+    private int mIconWidth = -1;
+    private int mIconHeight = -1;
+    private final Rect mOldBounds = new Rect();
+    private Canvas mCanvas = new Canvas();
+    private Context context;
+
+    public IconResizer(Context context) {
+        this.context = context;
+        mCanvas.setDrawFilter(new PaintFlagsDrawFilter(Paint.DITHER_FLAG,
+                Paint.FILTER_BITMAP_FLAG));
+
+        final Resources resources = context.getResources();
+        mIconWidth = mIconHeight = (int) resources.getDimension(
+                android.R.dimen.app_icon_size);
```

Comment:
Since it can be risky using a dimension outside of the app scope like `android.R.dimen.app_icon_size`, let's create a new `share_icon_size` dimension for the `mIconWidth` and `mIconHeight` variables.  The [touch target size](https://support.google.com/accessibility/android/answer/7101858) for buttons is 48dp and the custom share buttons (i.e. `ic_collaborate_48dp`, `ic_publish_48dp`, `ic_unpublish_48dp`, and `ic_wordpress_post_48dp`) are 48dp, so the `share_icon_size` dimension value should be `48dp` too.  See the code snippets below where _*Before*_ is the current implementation and _*After*_ is the proposed solution.

#### Before
```java
        final Resources resources = context.getResources();
        mIconWidth = mIconHeight = (int) resources.getDimension(
                android.R.dimen.app_icon_size);
```

#### After
```java
        mIconWidth = mIconHeight = (int) context.getResources().getDimension(R.dimen.share_icon_size);
```

We should also update the hard-coded size of the custom share buttons (i.e. `ic_collaborate_48dp`, `ic_publish_48dp`, `ic_unpublish_48dp`, and `ic_wordpress_post_48dp`) with the new `share_icon_size` dimension.